src/coreclr/jit/morph.cpp

```diff
@@ -13833,6 +13877,28 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
 //    division will be used, in that case this transform allows CSE to
 //    eliminate the redundant div from code like "x = a / 3; y = a % 3;".
 //
+//    Before:
+//        *  RETURN    int
```

Comment:
Something doesn't feel right in `Before:` tree. Did it mean to have `(V00 * V00) % V01`. Same in `After:`.

Reply:
As spoke offline, change `MUL` to `BINOP` and add a note that it is applicable for any `BINOP` that is safe to clone.